### PR TITLE
Unique Snap + changes to testing to fit this

### PR DESCRIPTION
**Pull request by christopher-hou**
app/controllers/teachers_controller.rb -- added Snap to error message
db/migrate/20210413081000_change_snap_unique.rb -- the migration
features/admin.feature and features/teacher.feature -- added Snap to each to make unique
features/form_submission.feature -- test if trying to use used Snap username prevents creating new teacher
features/step_definitions/form_steps.rb -- clarification on the test step

### CODE REVIEW

**Review by justinphyo — APPROVED** [review]
LGTM

File: features/admin.feature [hunk 1]
```diff
@@ -6,8 +6,8 @@ Feature: basic admin functionality
 
 Background: Has an Admin in DB
   Given the following teachers exist:
-  | first_name | last_name | admin | email                        |
-  | Joseph     | Mamoa     | true  | testadminuser@berkeley.edu   |
+  | first_name | last_name | admin | email                        | snap |
+  | Albert     | Admin     | true  | testadminuser@berkeley.edu   | au   |
```
Comment (tommywei110) [inline features/admin.feature]: Re discussion with Michael, should we add a test to make sure if we can still sign up without having a snap acc?

File: features/step_definitions/form_steps.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
 end
 
 And /^"(.*)" is not in the database$/ do |email|
-    Teacher.find_by(email: email).try(:destroy)
+    expect(Teacher.find_by(email: email).try(:destroy)).to be_nil
```
Comment (tommywei110) [inline features/step_definitions/form_steps.rb]: can we use `exists`? here? 
`expect(Teacher.exists?(email: email)).to be_false`
https://apidock.com/rails/ActiveRecord/Base/exists%3F/class

**Review by tommywei110 — APPROVED** [review]
Minor comments

File: features/admin.feature (hunk 1 above)
Comment (christopher-hou) [reply features/admin.feature]: Turns out that blank Snap is not getting saved as nil, like I expected, but rather as an empty string ('') which was causing an error. Wrote a new migration to fix it, set the default test Snap for teachers to '', and removed specifying Snap from a few of the tests that create multiple teachers to show that none of those are erroring which I hope should be good enough

File: features/step_definitions/form_steps.rb (hunk 2 above)
Comment (christopher-hou) [reply features/step_definitions/form_steps.rb]: Fixed!

**Pull request merged**